java/com/google/flatbuffers/ByteBufferReadBuf.java

```diff
@@ -0,0 +1,138 @@
+package com.google.flatbuffers;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class ByteBufferReadBuf implements ReadBuf {
```

Comment:
Most of this class is just an indirection, which can't help speed. Stick the `comparisonBuffer` in whoever uses the ByteBuffer?

Also the comparison functions are copied from aboved.. surely this can be structured better.

Reply:
I agree, maybe the comparison functions might be elsewhere. I will think about this on the final PR